packages/headless/src/api/analytics/product-listing-analytics.ts

```diff
@@ -0,0 +1,55 @@
+import {SearchPageClientProvider} from 'coveo.analytics';
+import {SearchEventRequest} from 'coveo.analytics/dist/definitions/events';
+import {
+  ConfigurationSection,
+  ProductListingSection,
+  SearchHubSection,
+} from '../../state/state-sections';
+import {getProductListingInitialState} from './../../features/product-listing/product-listing-state';
+import {BaseAnalyticsProvider} from './base-analytics';
+
+export type StateNeededByProductListingAnalyticsProvider =
+  ConfigurationSection & Partial<SearchHubSection & ProductListingSection>;
+
+export class ProductListingAnalyticsProvider
+  extends BaseAnalyticsProvider<StateNeededByProductListingAnalyticsProvider>
+  implements SearchPageClientProvider
+{
+  private initialState = getProductListingInitialState();
+
+  public getPipeline(): string {
+    return '';
+  }
+
+  public getSearchEventRequestPayload(): Omit<
+    SearchEventRequest,
+    'actionCause' | 'searchQueryUid'
+  > {
+    return {
+      queryText: '',
+      responseTime: 0,
+      results: this.mapResultsToAnalyticsDocument(),
+      numberOfResults: this.numberOfResults,
+    };
+  }
+
+  public getSearchUID() {
+    const newState = this.getState();
+    return newState.productListing?.responseId || this.initialState.responseId;
+  }
+
+  private mapResultsToAnalyticsDocument() {
+    return this.state.productListing?.products.map((p) => ({
+      documentUri: p.documentUri,
+      documentUriHash: p.documentUriHash,
+      permanentid: p.permanentid,
+    }));
+  }
+
+  private get numberOfResults() {
+    return (
+      this.state.productListing?.products.length ||
```

Comment:
The productListing attribute should [always be present](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/app/product-listing-engine/product-listing-engine.ts#L32) on the PLP engine. I think we can simplify here, and remove the `initialState` property.

Updating the type on line 11 to [ProductListingEngineState](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/app/product-listing-engine/product-listing-engine.ts#L34) should to the trick, but this might require moving the type to it's own file. Moving the `ProductListingSection` out of the `Partial` could be another option,

Reply:
I'm not sure I understand the first proposition here about the [ProductListingEngineState](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/app/product-listing-engine/product-listing-engine.ts#L34). You suggest to replace `ProductListingSection` by `ProductListingEngineState`  or something like
 type StateNeededByProductListingAnalyticsProvider = `ProductListingEngineState`

Reply:
Yea, maybe that could be one option, but if it's too complicated (e.g. due to circular references) then just moving `ProductListingSection` out of the `Partial` could be good too. I imagine the other uses-cases are doing this, so maybe best to stay consistent for now,